src/core/flow.js

```diff
@@ -44,12 +47,23 @@ export function Flow({
   validator,
   processMove,
   optimisticUpdate,
+  canMakeMove,
 }) {
   if (!ctx) ctx = () => ({});
   if (!events) events = {};
   if (!init) init = state => state;
   if (!validator) validator = () => true;
   if (!processMove) processMove = state => state;
+  if (!canMakeMove) {
+    canMakeMove = (G, ctx, playerID) => {
```

Comment:
`validator` above does a similar thing (deny a move if it's not valid in the current phase). How about we combine the two into a single `canMakeMove` function (also call this in reducer.js)?

Reply:
OK.